File: code/python/1400.py
# https://leetcode.com/problems/construct-k-palindrome-strings/
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        n = len(s)
        if n == k: return True
        if n < k: return False
        dict = Counter(s)
        count_odd = sum(x & 1 for x in dict.values())
        return count_odd <= k

File: code/cpp/1400.cpp
// https://leetcode.com/problems/construct-k-palindrome-strings/
class Solution {
public:
    bool canConstruct(string s, int k) {
        if (s.size() < k)
            return false;
        size_t freq[26] = {0}, cntOdd = 0;
        for (char& ch : s)
            freq[ch - 'a']++;
        for (size_t ele : freq)
            cntOdd += ele & 1;
        return cntOdd <= k;
    }
};
